Throw an error when Chromium download is failed

homaily opened the pull request:
Hi,

When Chromium download is failed (no internet connection for example) **downloadRevision** will not emit an error and therefore **onError** function in the install file will not be triggered and the user will not see the message:
> ERROR: Failed to download chromium ....




I'm using Node v6.11.2

homaily committed:
Throw an error when Chromium download is failed

aslushnikov commented:
This would play nice with #606 

Codewise, we should probably process.exit(1) to terminate the script with non-zero exit code

homaily commented:
We can do the **process.exit(1)** from the **onError** function

aslushnikov commented:
@homaily any updates on this? I'm very interested in this PR.

homaily commented:
@aslushnikov Sorry didn't get your point. Do you want the process.exit in the **onError** on int the place of **throw Error**?

aslushnikov commented on utils/ChromiumDownloader.js:
```diff
@@ -97,7 +97,7 @@ module.exports = {
-        .catch(err => err)
+        .catch(err => { throw err; })
         .then(() => {
           if (fs.existsSync(zipPath))
```
Unfortunately, there's no "finally" in promise chains, so I'd fake it like this:
```js
  .catch(err => err)
  .then(err => {
    if (fs.existsSync(zipPath))
      fs.unlinkSync(zipPath);
    if (err)
      throw err;
  });
```
This way we'll throw an error and cleanup zip if any

homaily replied:
Much better love it. I've updated my code to do that.

aslushnikov reviewed (commented):
Ah sorry, I got confused. I thought something else is going on here.

homaily committed:
Update ChromiumDownloader.js

Finishing the promise chain to remove the partially downloaded zip file (if exists)

aslushnikov reviewed (approved):
Thanks!

homaily commented:
You're welcome Andrey. Appreciated.
